### PR TITLE
Attribute `cors_allowed_host_names` added for cloudfront resource.

### PR DESCRIPTION
**Pull request by tahir-duplo**
Attribute `cors_allowed_host_names` added for cloudfront resource.

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-4333 Add CORS allowed host support in cloudfront distribution.](https://app.clickup.com/t/8655600/DUPLO-4333)

File: duplocloud/resource_duplo_aws_cloudfront_distribution.go [hunk 1]
```diff
@@ -840,7 +846,7 @@ func resourceAwsCloudfrontDistributionRead(ctx context.Context, d *schema.Resour
 	}
 
 	flattenAwsCloudfrontDistribution(d, duplo.Distribution.DistributionConfig)
-
+	// d.Set("cors_allowed_host_names", flattenStringList("No Data From Backend"))
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_aws_cloudfront_distribution.go:849]: @tahir-duplo  - This commented-out line seems to have been the only place where we will have read this setting from the backend.  If we do not read the setting from a backend, I think we may have an issue with this code.

File: duplocloud/resource_duplo_aws_cloudfront_distribution.go [hunk 2]
```diff
@@ -911,10 +921,11 @@ func resourceAwsCloudfrontDistributionUpdate(ctx context.Context, d *schema.Reso
 	updateS3OAI(duplo.Distribution.DistributionConfig, rq)
 
 	resp, err := c.AwsCloudfrontDistributionUpdate(tenantID, &duplosdk.DuploAwsCloudfrontDistributionCreate{
-		Id:                 cfdId,
-		DistributionConfig: rq,
-		IfMatch:            d.Get("etag").(string),
-		UseOAIIdentity:     d.Get("use_origin_access_identity").(bool),
+		Id:                   cfdId,
```
Comment (joek-duplo) [inline duplocloud/resource_duplo_aws_cloudfront_distribution.go:924]: @tahir-duplo  - does this work when the user does not specify any value?  In other words, does `expandStringList` handle getting an empty list?  Second question:  can `d.Get("cors_allowed_host_names")` return `nil`?  if so, we will get a panic at runtime.

**Review by joek-duplo — CHANGES_REQUESTED** [review]

File: duplocloud/resource_duplo_aws_cloudfront_distribution.go (hunk 1 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_aws_cloudfront_distribution.go:849]: Backend just expect CORS host names on create API but does not return this data as part of get API

File: duplocloud/resource_duplo_aws_cloudfront_distribution.go [hunk 3]
```diff
@@ -860,7 +866,11 @@ func resourceAwsCloudfrontDistributionCreate(ctx context.Context, d *schema.Reso
 	c := m.(*duplosdk.Client)
 
 	rq := expandAwsCloudfrontDistributionConfig(d)
-	resp, err := c.AwsCloudfrontDistributionCreate(tenantID, &duplosdk.DuploAwsCloudfrontDistributionCreate{DistributionConfig: rq, UseOAIIdentity: d.Get("use_origin_access_identity").(bool)})
+	resp, err := c.AwsCloudfrontDistributionCreate(tenantID, &duplosdk.DuploAwsCloudfrontDistributionCreate{
+		DistributionConfig:   rq,
+		UseOAIIdentity:       d.Get("use_origin_access_identity").(bool),
+		CorsAllowedHostNames: expandStringList(d.Get("cors_allowed_host_names").([]interface{})),
```
[the remark this answers is not in the document]
Comment (tahir-duplo) [reply duplocloud/resource_duplo_aws_cloudfront_distribution.go:872]: Yes it work, it returns empty list.

File: duplocloud/resource_duplo_aws_cloudfront_distribution.go (hunk 2 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_aws_cloudfront_distribution.go:924]: Yes, it works, it returns an empty list.

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**